src/lib/nextWordGuesserStrategy.ts

```diff
@@ -16,57 +16,85 @@
+ *  const wordList = new wordList(methodThatReturnsAListOfWords())
+ *  const asciiValueGuesser = new AsciiValueGuesserStrategy(wordList);
+ *  const asciiValueGuesserAfterGuessingHELLO = asciiValueGuesser.withPreviousGuess({
+ *      word: 'HELLO',
+ *      result: someFunctionThatYieldsExclusionAndMandateRulesFromGuess('HELLO')
+ *  });
+ *  expect(asciiValueGuesserAfterGuessingHELLO.getAlreadyGuessedLetters()).toEqual(['H', 'E', 'L', 'O']);
+ *  console.log('Next optimal word guess after HELLO is: ',
+ *      asciiValueGuesserAfterGuessingHELLO.guessNextWordAndScore());
+ * ```
  */
-
-export interface INextWordGuesserStrategyBase {
-    getAlreadyGuessedLetters(): string[];
-
-    guessNextWordAndScore(): WordGuessAndScore;
-
-    withPreviousGuess(candidate: WordGuessAndResult): this;
-
-    get previousGuesses(): WordGuessAndResult[];
-    get wordList(): WordList;
-}
-
-export interface IStrategyScoreMethod {
-    scoreForGuess(guess: string): number;
-}
-
-export abstract class NextWordGuesserStrategyBase implements INextWordGuesserStrategyBase {
+export default abstract class NextWordGuesserStrategyBase {
+    /**
+     * Track the previously-guessed words and their resulting exclusion/mandate rules.
+     */
     protected myPreviousGuesses: WordGuessAndResult[] = [];
 
+    /**
+     *
+     * @param myWordList - the `WordList` object to use for choosing the next optimal guess.
+     */
     constructor(protected myWordList: WordList) {}
 
+    /**
+     * Implement this with a metric for scoring for the given guess, determined by the guessing strategy in use.
+     *
+     * @param guess - The guess candidate word.
+     *
+     * @returns A score calculated by a given guessing strategy, roughly equal to the number of letter & position
+     *  combinations' worth of information that would be given by this guess in that strategy.
+     */
     abstract scoreForGuess(guess: string): number;
 
-    withPreviousGuess(candidate: WordGuessAndResult) {
+    /**
+     * Modifies the calling object to procss the given candidate guess and store its resulting rules, such as certain
+     *  letters being mandated at or excluded from certain positions.
+     *
+     * @param candidate - A pairing of the guessed word and its resulting `LetterAtPositionInWordRule` rules.
+     *
+     * @returns `this`, after having processed the previous guess and its resulting rules.
+     */
+    withPreviousGuess(candidate: WordGuessAndResult): this {
         this.myWordList.processExclusionsFromRules(candidate.result);
         this.myPreviousGuesses.push(candidate);
         return this;
     }
 
+    /**
+     * Calculates the score for all remaining possible candidate words and then returns the highest-scoring one.
+     *
+     * @returns The next optimal word to guess for this particular strategy.
+     *
+     * @throws `NoMoreGuessesError`
+     * When there are no more guesses possible (i.e., the list of candidate words becomes empty).
```

Comment:
Nit: Consistency of casing.
```suggestion
     * when there are no more guesses possible (i.e., the list of candidate words becomes empty).
```